### PR TITLE
Add initial support for MySQL databases

### PR DESCRIPTION
**Pull request by kuuttia**
- Clone not implemented yet
- Update id sequences not implemented yet

### CODE REVIEW

File: lib/MySqlDatabaseManager.js [hunk 1]
```diff
@@ -1,16 +1,185 @@
...
+  this._cachedTableNames = null;
 }
 
 classUtils.inherits(MySqlDatabaseManager, DatabaseManager);
 
+/**
+ * @Override
+ */
+MySqlDatabaseManager.prototype.createDbOwnerIfNotExist = function() {
+  return this._masterQuery("CREATE USER IF NOT EXISTS ?@'localhost' IDENTIFIED BY ?", [this.config.knex.connection.user, this.config.knex.connection.password])
```
Comment (elhigu) [inline lib/MySqlDatabaseManager.js]: I don't know much about mysql, so I have to ask... why there is hard coded 'localhost' when creating user?

File: lib/MySqlDatabaseManager.js [hunk 2]
```diff
@@ -1,16 +1,185 @@
...
+    this._updateTableNameCache(knex, config);
+  }
+
+  return this._cachedTableNames.then(function (tableNames) {
+    if (!_.isEmpty(tableNames)) {
+      return knex.transaction(function(trx) {
+        return knex.raw('SET FOREIGN_KEY_CHECKS = 0').transacting(trx)
+          .then(function() {
+            return Promise.map(tableNames, function(tableName) {
+              return knex.raw('TRUNCATE TABLE ' + tableName).transacting(trx);
```
Comment (elhigu) [inline lib/MySqlDatabaseManager.js]: does this also reset the sequences? IIRC with knex one can do it like this `trx(tableName).truncate()`.

File: .travis.yml [hunk 3]
```diff
@@ -24,6 +24,7 @@ before_install:
   - wget https://raw.githubusercontent.com/Vincit/travis-oracledb-xe/master/accept_the_license_agreement_for_oracledb_xe_11g_and_install.sh
   - bash ./accept_the_license_agreement_for_oracledb_xe_11g_and_install.sh
   - npm install oracledb
+  - docker-compose up -d
```
Comment (elhigu) [inline .travis.yml]: thats nice!

File: lib/MySqlDatabaseManager.js (hunk 2 above)
Comment (kuuttia) [inline lib/MySqlDatabaseManager.js]: Yes MySQL truncate resets auto_increment sequences "Any AUTO_INCREMENT value is reset to its start value. This is true even for MyISAM and InnoDB, which normally do not reuse sequence values".

**Pull request merged**